src/caffe/layers/detectnet_transform_layer.cpp

```diff
@@ -439,7 +439,14 @@ void DetectNetTransformationLayer<Dtype>::transform_scale(
     Dtype scale_x = (Dtype)size.width / img.cols;
     Dtype scale_y = (Dtype)size.height / img.rows;
 
-    resize(img, *img_temp, size, cv::INTER_CUBIC);
+    // Resize using bicubic interpolation and then clamp to [min, max]
+    double minval, maxval;
+    cv::minMaxLoc(img, &minval, &maxval);
+    cv::resize(img, *img_temp, size, 0, 0, cv::INTER_CUBIC);
+    cv::threshold(*img_temp, *img_temp, maxval, 0, cv::THRESH_TRUNC);
+    *img_temp *= -1;
+    cv::threshold(*img_temp, *img_temp, minval, 0, cv::THRESH_TRUNC);
```

Comment:
I might have misunderstood something but since you have multiplied the array by `-1` shouldn't you use `-minval` for the threshold?